vsphere/resource_vsphere_role.go

```diff
@@ -0,0 +1,142 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-provider-vsphere/vsphere/internal/helper/structure"
+	"github.com/vmware/govmomi/object"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func resourceVsphereRole() *schema.Resource {
+	sch := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the storage policy.",
```

Comment:
How are duplicate role names managed? Have you tested it?

Reply:
Yes, duplicate name will throw an error.